lib/_stream_writable.js

```diff
@@ -498,22 +502,41 @@ function onwrite(stream, er) {
     }
 
     if (sync) {
-      process.nextTick(afterWrite, stream, state, cb);
+      // It is a common case that the callback passed to .write() is always
+      // the same. In that case, we do not schedule a new nextTick(), but rather
+      // just increase a counter, to improve performance and avoid memory
+      // allocations.
+      if (state.afterWriteTickInfo !== null &&
+          state.afterWriteTickInfo.cb === cb) {
+        state.afterWriteTickInfo.count++;
+      } else {
+        state.afterWriteTickInfo = { count: 1, cb, stream, state };
+        process.nextTick(afterWriteTick, state.afterWriteTickInfo);
+      }
     } else {
-      afterWrite(stream, state, cb);
+      afterWrite(stream, state, 1, cb);
     }
   }
 }
 
-function afterWrite(stream, state, cb) {
+function afterWriteTick({ stream, state, count, cb }) {
```

Comment:
This might clear the wrong object. I think clearing the count and cb is safer?

Reply:
Not sure if it matter though

Reply:
This comment is for the row below.

Reply:
@ronag So … the effect of setting `afterWriteTickInfo` to `null` is that the next time the code above is reached, a new `process.nextTick()` call with a new `afterWriteTickInfo` object is made. That’s always safe, right?

I think setting `.cb` to null would have the same effect, and `.count` is cleared anyway. I can do that instead, if you prefer, although it might screw with the map/hidden class of `afterWriteTickInfo`, as `.cb` is always a function right now.

Reply:
I was more thinking of the case where you have two different cbs, e.g.

```
js
write('a', cba) // schedule tick 1
write('b', cbb) // clear info a, schedule tick b
// ...
// cba
// clear info b
// cbb
// clear nothing
```

Probably not a problem but a little weird...

Reply:
>  it might screw with the map/hidden class

Oh, I didn't know that `null` could cause a problems with that once it's been a function type.

Reply:
> The a tick will actually clear the info for the b tick.

> Probably not a problem, but maybe a little weird... I don't have a strong opinion if you think it's fine.

Yeah, I think that’s fine, because it would only make a difference if there’s a `write('b', cbb)` inside `cba()`, and that seems like a somewhat unlikely scenario, and even then it would only affect performance, not behaviour.